mutex.go

```diff
@@ -45,14 +45,15 @@ var (
 func Lock(l sync.Locker, d time.Duration) error {
 	done := make(chan struct{}, 1)
 	m, ok := l.(*mutex)
+	var err error = nil
 	// if l is a mutex, lock with a timeout context.
 	if ok {
-		ctx, cancel := context.WithTimeout(context.Background(), d*time.Second)
+		ctx, cancel := context.WithTimeout(context.Background(), d)
 		defer cancel()
-		go func(context.Context) {
-			m.lock(ctx)
+		go func() {
+			err = m.lock(ctx)
 			done <- struct{}{}
-		}(ctx)
+		}()
 	} else {
 		go func() {
 			l.Lock()
```

Comment:
The issue with this `else` part is it seems quite useless: a regular `sync.Locker` mutex may never return and so the promise of the `func Lock(l sync.Locker, d time.Duration)` func is actually never fulfilled. In other words, it won't return after the `d` timeout.

I wonder if it should be considered a programming error, passing in a `sync.Locker` which is not of type `(*mutex)`, and this code should panic, indicating this should never have happened.

But then, going further, I actually wonder why then stick to `sync.Locker` interface, if this code will never accept a regular non-gcslock `sync.Locker` mutex.

WDYT?